src/executioners/SpectralExecutionerLinearElastic.C

```diff
@@ -269,31 +198,29 @@ SpectralExecutionerLinearElastic::hasStressConverged(const FFTBufferBase<RankTwo
   Complex error_n(0.0, 0.0);
   Complex error_0(0.0, 0.0);
 
-  // error
+  // Error: Ensure overall equilibrium
   for (int freq_x = 0; freq_x < ni; ++freq_x)
     for (int freq_y = 0; freq_y < nj; ++freq_y)
       for (int freq_z = 0; freq_z < nk; ++freq_z)
       {
-        const std::array<Complex, 3> freq{ivec[freq_x] * I, jvec[freq_y] * I, kvec[freq_z] * I};
-
-        std::array<Complex, 3> kvector_stress;
-        kvector_stress[0] = stress.reciprocalSpace()[index](0, 0) * freq[0] +
-                            stress.reciprocalSpace()[index](1, 0) * freq[1] +
-                            stress.reciprocalSpace()[index](2, 0) * freq[2];
-        kvector_stress[1] = stress.reciprocalSpace()[index](0, 1) * freq[0] +
-                            stress.reciprocalSpace()[index](1, 1) * freq[1] +
-                            stress.reciprocalSpace()[index](2, 1) * freq[2];
-        kvector_stress[2] = stress.reciprocalSpace()[index](0, 2) * freq[0] +
-                            stress.reciprocalSpace()[index](1, 2) * freq[1] +
-                            stress.reciprocalSpace()[index](2, 2) * freq[2];
-
-        error_n += kvector_stress[0] * kvector_stress[0] + kvector_stress[1] * kvector_stress[1] +
-                   kvector_stress[1] * kvector_stress[1];
+        const ComplexVectorValue freq{ivec[freq_x] * I, jvec[freq_y] * I, kvec[freq_z] * I};
+
+        ComplexVectorValue kvector_stress;
+        kvector_stress(0) = stress.reciprocalSpace()[index](0, 0) * freq(0) +
+                            stress.reciprocalSpace()[index](1, 0) * freq(1) +
+                            stress.reciprocalSpace()[index](2, 0) * freq(2);
+        kvector_stress(1) = stress.reciprocalSpace()[index](0, 1) * freq(0) +
+                            stress.reciprocalSpace()[index](1, 1) * freq(1) +
+                            stress.reciprocalSpace()[index](2, 1) * freq(2);
+        kvector_stress(2) = stress.reciprocalSpace()[index](0, 2) * freq(0) +
+                            stress.reciprocalSpace()[index](1, 2) * freq(1) +
+                            stress.reciprocalSpace()[index](2, 2) * freq(2);
+
+        error_n += kvector_stress(0) * kvector_stress(0) + kvector_stress(1) * kvector_stress(1) +
+                   kvector_stress(2) * kvector_stress(2);
```

Comment:
```suggestion
        error_n += kvector_stress.norm_sq();
```

Reply:
`.norm_sq()` takes the norm of each component's complex number, not what we look for here.

Reply:
Oh, it would be the component times its complex conjugate rather than component square? Thats really not what is wanted here?

Reply:
`norm_sq` obtains the norm of each of the components and sums them up. The error formula I am basing this off takes the norm of the entire vector. Results are different.

There is more work to do in terms of convergence checks. The formula implemented here captures stress equilibrium. But we could add BC error and others. We can let all this for next revision.